### PR TITLE
Fix discard config panel remove layer

### PR DESCRIPTION
**Pull request by junqiu-lei**
Signed-off-by: Junqiu Lei <junqiu@amazon.com>

### Description
Fix discard config panel remove layer issue.

### Issue Reproduce
1. Add new layer
2. Complete configuration and click update
3. Reopen this layer config panel
4. Click discard
5. This layer will be removed from layers

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VijayanB — APPROVED** [review]

File: maps_dashboards/public/components/layer_config/layer_config_panel.tsx [hunk 1]
```diff
@@ -74,6 +74,9 @@ export const LayerConfigPanel = ({
   const onUpdate = () => {
     updateLayer();
     closeLayerConfigPanel(false);
+    if (isNewLayer) {
```
Comment (heemin32) [inline maps_dashboards/public/components/layer_config/layer_config_panel.tsx:77]: Why set false when isNewLayer is true?

Should it be ?
```suggestion
    setIsNewLayer(isNewLayer)
```

Comment (junqiu-lei) [reply maps_dashboards/public/components/layer_config/layer_config_panel.tsx:77]: Every time we create new layer, the `isNewLayer ` will be true,  after clicking update, we set it to false. Adding if check here can avoid no need trigger `setIsNewLayer(isNewLayer)` when it's not  a new layer.

Comment (heemin32) [reply maps_dashboards/public/components/layer_config/layer_config_panel.tsx:77]: Thanks for the explanation. Maybe having a common function to unset isNewLayer value to make it easier to understand? 
```
markAsOldLayer() // I cannot come up with a good name here...

markAsOldLayer(() => {
    if(isNewLayer) {
        setIsNewLayer(false);
    }
  });
```
I will leave it to you though as current implementation might be easy enough to someone who is familiar with react.

**Review by heemin32 — APPROVED** [review]

File: maps_dashboards/public/components/layer_config/layer_config_panel.tsx (hunk 1 above)
Comment (junqiu-lei) [reply maps_dashboards/public/components/layer_config/layer_config_panel.tsx:77]: Thanks @heemin32's suggestion , I'll do it later

**Pull request merged**